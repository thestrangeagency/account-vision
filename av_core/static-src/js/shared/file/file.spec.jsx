import React from 'react';
import { shallow } from 'enzyme';

import File from './file';

const onClickDelete = jest.fn();
const onClickEdit = jest.fn();

const fileProps = {
  date: '10/28/2017',
  name: 'my-file.txt',
  onClickDelete,
  onClickEdit,
  size: '200kb',
  type: 'txt',
};

const wrapper = shallow(<File {...fileProps} />);

test('it renders', () => {
  expect(wrapper).toMatchSnapshot();
});

test('it calls remove callback', () => {
  wrapper.find('[aria-label="Remove"]').simulate('click');
  expect(onClickDelete).toHaveBeenCalled();
});

test('it calls edit callback', () => {
  wrapper.find('[aria-label="Edit"]').simulate('click');
  expect(onClickEdit).toHaveBeenCalled();
});
