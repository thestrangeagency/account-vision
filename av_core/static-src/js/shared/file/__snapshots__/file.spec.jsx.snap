// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders 1`] = `
<div
  className="file-upload d-flex align-items-center"
>
  <div
    className="mx-2"
  >
    <i
      className="far fa-file fa-lg"
    />
  </div>
  <div
    className="md-mx-2 truncate w-25"
  >
    my-file.txt
  </div>
  <div
    className="ml-2 md-mx-2 d-none d-md-block file-upload__size text-nowrap truncate"
  >
    200kb
  </div>
  <div
    className="mx-2"
  >
    10/28/2017
  </div>
  <div>
    <button
      aria-label="Edit"
      className="px-2 far fa-edit fa-lg file-upload__button"
      onClick={[Function]}
      type="button"
    />
    <button
      aria-label="Remove"
      className="far fa-times-circle fa-lg text-danger-light file-upload__button"
      onClick={[Function]}
      type="button"
    />
  </div>
</div>
`;
